src/main/java/software/amazon/kinesis/connectors/flink/internals/publisher/fanout/FanOutShardSubscriber.java

```diff
@@ -321,7 +321,11 @@ public void onError(Throwable throwable) {
 			// Cancel the subscription to signal the onNext to stop requesting data
 			cancelSubscription();
 
-			subscriptionErrorEvent.set(new SubscriptionErrorEvent(throwable));
+			if (subscriptionErrorEvent.get() == null) {
+				subscriptionErrorEvent.set(new SubscriptionErrorEvent(throwable));
+			} else {
+				LOG.warn("Previous error passed to consumer for processing. Ignoring subsequent exception.", throwable);
```

Comment:
As the previous exception must have cancelled the subscription, subsequent ErrorEvents appear to be very unusual. We could consider making that more apparent in the log message.